### PR TITLE
Cloudstack securityrule test refactor pr one

### PR DESCRIPTION
**Pull request by fgan1**
# Description
Refactoring the Cloudstack Security Rules Plugin context. 

# Proposed changes
- Refactoring in the Cloudstack Security Rules Plugin

# Additional information
- This branch depends on a branch in the Fogbow Common (cloudstack-securityrules-test-refator)
Note: Maybe happen compiling error at PRs before PR5 in regards to Common Project because there were some changes in the PR5.

# PR Slices
[x] One: Refactoring in the Cloudstack Security Rules Plugin
[] Two: Migrating Cloudstack asynchronous operation to the Cloudstack  Utils
[] Three: Refactoring "request Instance" tests context.
[] Four: Refactoring "get Instance" tests context.
[] Five: Refactoring "delete instance" tests context.

# Reminding
- PRs order: 
develop < PR1 < PR2 < PR3 < PR4 < PR5

### CODE REVIEW

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java [hunk 1]
```diff
@@ -6,154 +6,181 @@
...
-		List<ListFirewallRulesResponse.SecurityRuleResponse> securityRulesResponse = response.getSecurityRulesResponse();
-		return convertToFogbowSecurityRules(securityRulesResponse);
+			String jsonResponse = this.client.doGetRequest(uriRequest.toString(), cloudUser);
+            ListFirewallRulesResponse response = ListFirewallRulesResponse.fromJson(jsonResponse);
+            List<ListFirewallRulesResponse.SecurityRuleResponse> securityRulesResponse =
+                    response.getSecurityRulesResponse();
+            return convertToFogbowSecurityRules(securityRulesResponse);
+        } catch (HttpResponseException e) {
+            throw CloudStackHttpToFogbowExceptionMapper.get(e);
+        }
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: Fix the indentation of this method.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java [hunk 2]
```diff
@@ -163,14 +190,17 @@ protected String waitForDeleteResult(CloudStackHttpClient client, String jobId,
 			String ipAddress = securityRuleResponse.getIpAddress();
 			SecurityRule.EtherType etherType = inferEtherType(ipAddress);
 			SecurityRule.Protocol protocol = getFogbowProtocol(securityRuleResponse.getProtocol());
+            String instanceId = securityRuleResponse.getInstanceId();
 
-            SecurityRuleInstance securityRuleInstance = new SecurityRuleInstance(securityRuleResponse.getInstanceId(), direction, portFrom, portTo, cidr, etherType, protocol);
+            SecurityRuleInstance securityRuleInstance = new SecurityRuleInstance(instanceId, direction,
+                    portFrom, portTo, cidr, etherType, protocol);
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: Fix the indentation of this method.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePluginTest.java [hunk 3]
```diff
@@ -41,7 +42,7 @@
 @RunWith(PowerMockRunner.class)
 @PowerMockIgnore({"javax.net.ssl.*", "javax.crypto.*" })
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePluginTest.java]: Maybe you should turn these strings into constants.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java (hunk 1 above)
Comment (fgan1) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: I've fixed this in the PR five.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java (hunk 2 above)
Comment (fgan1) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: I've fixed this in the PR five.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePluginTest.java (hunk 3 above)
Comment (fgan1) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePluginTest.java]: It was removed.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java (hunk 1 above)
Comment (fgan1) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: My IDE wasn't catching this indentation mistake...
Has anyone ever had this problem before?

Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/securityrule/v4_9/CloudStackSecurityRulePlugin.java]: Apparently IDEs don't show these indentation errors, but you can set them not to use tab spaces and replace these tabs with 4 spaces.

**Pull request merged**